syntax = "proto3";
package product;

service ProductService {
  rpc AddProduct (AddProductRequest) returns (AddProductResponse);
  rpc AddCategory (AddCategoryRequest) returns (AddCategoryResponse);
  rpc DeleteProduct (DeleteProductRequest) returns (DeleteProductResponse);
}

service UserService {
  rpc RegisterUser (RegisterUserRequest) returns (RegisterUserResponse);
  rpc AuthenticateUser (AuthenticateUserRequest) returns (AuthenticateUserResponse);
}

message Product {
  int32 id = 1;
  string name = 2;
  string description = 3;
  double price = 4;
  int32 category_id = 5;
}

message Category {
  int32 id = 1;
  string name = 2;
}

message AddProductRequest {
  Product product = 1;
}

message AddProductResponse {
  int32 id = 1;
}

message AddCategoryRequest {
  Category category = 1;
}

message AddCategoryResponse {
  int32 id = 1;
}

message DeleteProductRequest {
  int32 id = 1;
}

message DeleteProductResponse {
  bool success = 1;
}

message RegisterUserRequest {
  string username = 1;
  string email = 2;
  string password = 3;
}

message RegisterUserResponse {
  int32 id = 1;
}

message AuthenticateUserRequest {
  string username = 1;
  string password = 2;
}

message AuthenticateUserResponse {
  string token = 1;
}
